File: sliding-window-pattern/maxsumsub.js
function maxSubarraySum(arr, n) {
  if (n > arr.length) {
    return null;
  }

  let maxSum = 0;
  let tempSum = 0;
  for (let i = 0; i < n; i++) {
    maxSum += arr[i];
  }

  tempSum = maxSum;
  for (let i = n; i < arr.length; i++) {
    tempSum = tempSum - arr[i - n] + arr[i];
    maxSum = Math.max(maxSum, tempSum);
  }
  return maxSum;
}

maxSubarraySum([2,6,9,2,1,8,5,6,3], 3)

File: sliding-window-pattern/maxsumsub.py
def max_sub_sum(arr, n):
    if len(arr) < n:
        return None

    max_sum = 0
    for i in range(n):
        max_sum += arr[i]

    temp_sum = max_sum
    for i in range(n, len(arr)):
        temp_sum = temp_sum - arr[i - n] + arr[i]
        max_sum = max(max_sum, temp_sum)
    return max_sum

print(max_sub_sum([2,6,9,2,1,8,5,6,3], 3))